### PR TITLE
Refactor stylesheets

### PR DESCRIPTION
**Pull request by mikefowler**
Refactored stylesheets and view markup for maintainability and to remove use of BassCSS stylesheet. Also implemented responsive styles.

### CODE REVIEW

**Comment by mikefowler** [comment]
@jonmagic seems like a big commit, but it's almost exclusively limited to stylesheet and view changes.

**Comment by jonmagic** [comment]
Sweet! I'll pull this down tonight and kick the tires.

**Comment by jonmagic** [comment]
The form on the home page:

coderdojo - san francisco-4

Looks a bit different than the forms on the event page:

coderdojo - san francisco-5

coderdojo - san francisco-6

**Comment by jonmagic** [comment]
Looks great :+1: Everything feels more consistent. I do have one question about the class names, I've never seen the pattern of using double underscore `__` before, where does that come from?

File: app/views/lessons/_lesson.html.erb [hunk 1]
```diff
@@ -1,22 +1,63 @@
...
+      <li><strong>Time:</strong> <%= lesson.upcoming_event.start_time.strftime("%l:%M%p") %> - <%= lesson.upcoming_event.end_time.strftime("%l:%M%p") %></li>
+      <li><strong>Location:</strong> <a href="https://maps.google.com/maps?client=safari&q=88+Colin+P+Kelly+Jr+St+San+Francisco,+CA+94107&ie=UTF-8&hq=&hnear=0x80858077fe6ae953:0x1cf829f1248fb8f4,88+Colin+P+Kelly+Jr+St,+San+Francisco,+CA+94107&gl=us&ei=A5LHUrnaB8vqoATK9IHgAQ&ved=0CCwQ8gEwAA" target="_blank">88 Colin P Kelly Jr St, San Francisco, CA »</a></li>
+      <li><strong>Teacher:</strong> <%= link_to lesson.upcoming_event.teacher_github_username, "https://github.com/#{lesson.upcoming_event.teacher_github_username}" %></li>
+    </ul>
     <% end %>
+
   </div>
-</div>
+  
+  <% if @show_registration && @lesson.upcoming_event.present? %>
```
Comment (jonmagic) [inline app/views/lessons/_lesson.html.erb:28]: This `@show_registration` is interesting, this opens up the possibility of showing the form on the home page now too right? That might be nice, why make folks click on the lesson to register when they could register right from the homepage.

That would put two forms on the homepage though (this one and the newsletter) so maybe that isn't ideal, but I like having the option to try it at least by just setting `@show_registration` to true.

Comment (mikefowler) [inline app/views/lessons/_lesson.html.erb:28]: Correct. I hadn't thought about it from the perspective of making the form available on the home page as much as just reducing template duplication, but definitely a nice side effect :+1:

**Comment by mikefowler** [comment]
@jonmagic: The class naming convention is [BEM notation](http://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/). Mostly just a habit for me at this point, it really helps me think about and read CSS in a more object-oriented way.

I'll make those forms more consistent (good catch) and update the PR.

**Comment by mikefowler** [comment]
@jonmagic, go ahead and merge if everything looks cool with that additional commit. :ship:

**Pull request merged**